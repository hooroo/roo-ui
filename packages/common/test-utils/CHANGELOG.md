# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.4.0"></a>
# [0.4.0](https://github.com/hooroo/roo-ui/compare/v0.3.3...v0.4.0) (2018-04-26)


### Features

* add theming ([bee896a](https://github.com/hooroo/roo-ui/commit/bee896a))
