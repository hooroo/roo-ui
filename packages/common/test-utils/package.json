{
  "name": "@roo-ui/test-utils",
  "version": "0.0.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
