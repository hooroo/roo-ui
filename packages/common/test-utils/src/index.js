export { default as shallowWithTheme } from './shallowWithTheme'; // eslint-disable-line import/prefer-default-export
