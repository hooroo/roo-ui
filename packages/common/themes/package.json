{
  "name": "@roo-ui/themes",
  "version": "0.4.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
