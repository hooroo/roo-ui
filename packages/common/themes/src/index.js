import qantas from './qantas';

export { qantas }; // eslint-disable-line import/prefer-default-export
