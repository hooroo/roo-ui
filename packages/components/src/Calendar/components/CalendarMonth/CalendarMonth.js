import { themeGet } from 'styled-system';
import { css } from 'styled-components';

import { Box } from '../../../';

const CalendarMonth = Box.extend`
  text-align: center;
  padding: 0 ${themeGet('space.4')};
  width: 100%;

  ${props => props.monthsToDisplay === 2 &&
    css`
      width: ${100 / 2}%;
    `};

  ${props => props.monthsToDisplay === 3 &&
    css`
      width: ${100 / 3}%;
    `};
`;

CalendarMonth.defaultProps = {
  ...Box.defaultProps,
  blacklist: [...Object.keys(Box.propTypes), 'monthsToDisplay'],
};

export default CalendarMonth;
