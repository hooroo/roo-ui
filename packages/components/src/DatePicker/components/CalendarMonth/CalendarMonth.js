import React from 'react';
import PropTypes from 'prop-types';
import { themeGet } from 'styled-system';
import { css } from 'styled-components';
import { Box, Text } from '../../../';
import CalendarWeekdays from '../CalendarWeekdays';
import CalendarDays from '../CalendarDays';
import { styledOmitProps } from '../../../../lib';

const MonthWrapper = styledOmitProps(Box, { omit: ['monthsToDisplay', 'stacked'] })`
  text-align: center;
  padding: 0 ${themeGet('space.4')};
  width: ${props => `${100 / props.monthsToDisplay}%`};

  ${props => props.stacked &&
    css`
      width: 100%;
      margin-top: ${themeGet('space.8')};

      &:first-of-type {
        margin-top: 0;
      }
    `};
`;

MonthWrapper.displayName = 'MonthWrapper';

const CalendarMonth = ({
  monthsToDisplay, month, monthName, year, stacked, weekdayNames,
  weeks, getDateProps, disabledDates, interactiveDisabledDates,
  onMouseEnterOfDay, isInRange,
}) => (
  <MonthWrapper monthsToDisplay={monthsToDisplay} stacked={stacked}>
    <Text textStyle="caps">
      {monthName} {year}
    </Text>
    <CalendarWeekdays month={month} year={year} weekdayNames={weekdayNames} />
    <CalendarDays
      weeks={weeks}
      month={month}
      year={year}
      getDateProps={getDateProps}
      disabledDates={disabledDates}
      interactiveDisabledDates={interactiveDisabledDates}
      onMouseEnterOfDay={onMouseEnterOfDay}
      isInRange={isInRange}
    />
  </MonthWrapper>
);

MonthWrapper.defaultProps = {
  ...Box.defaultProps,
};

CalendarMonth.defaultProps = {
  monthsToDisplay: 1,
  stacked: false,
  disabledDates: [],
  interactiveDisabledDates: false,
  isInRange: () => {},
  onMouseEnterOfDay: () => {},
};

CalendarMonth.propTypes = {
  month: PropTypes.number.isRequired,
  year: PropTypes.number.isRequired,
  monthName: PropTypes.string.isRequired,
  weeks: PropTypes.arrayOf(PropTypes.array).isRequired,
  weekdayNames: PropTypes.arrayOf(PropTypes.string).isRequired,
  disabledDates: PropTypes.arrayOf(PropTypes.instanceOf(Date)),
  interactiveDisabledDates: PropTypes.bool,
  monthsToDisplay: PropTypes.number,
  stacked: PropTypes.bool,
  getDateProps: PropTypes.func.isRequired,
  onMouseEnterOfDay: PropTypes.func,
  isInRange: PropTypes.func,
};

export default CalendarMonth;
