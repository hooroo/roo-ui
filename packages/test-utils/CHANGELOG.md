# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.50.7"></a>
## [0.50.7](https://github.com/hooroo/roo-ui/compare/v0.50.6...v0.50.7) (2018-08-14)


### Bug Fixes

* pass theme to child components; update snapshots ([#153](https://github.com/hooroo/roo-ui/issues/153)) ([1a83477](https://github.com/hooroo/roo-ui/commit/1a83477))




<a name="0.39.1"></a>
## [0.39.1](https://github.com/hooroo/roo-ui/compare/v0.39.0...v0.39.1) (2018-06-26)




**Note:** Version bump only for package @roo-ui/test-utils

<a name="0.39.0"></a>
# [0.39.0](https://github.com/hooroo/roo-ui/compare/v0.38.4...v0.39.0) (2018-06-26)


### Features

* add build:watch scripts ([#98](https://github.com/hooroo/roo-ui/issues/98)) ([bcae0bf](https://github.com/hooroo/roo-ui/commit/bcae0bf))




<a name="0.32.0"></a>
# [0.32.0](https://github.com/hooroo/roo-ui/compare/v0.31.0...v0.32.0) (2018-06-15)


### Features

* add autocomplete component ([#77](https://github.com/hooroo/roo-ui/issues/77)) ([eee3f8c](https://github.com/hooroo/roo-ui/commit/eee3f8c))




<a name="0.19.0"></a>
# [0.19.0](https://github.com/hooroo/roo-ui/compare/v0.18.0...v0.19.0) (2018-05-22)


### Features

* Add <List/> & <ListItem> ([#49](https://github.com/hooroo/roo-ui/issues/49)) ([b922a8a](https://github.com/hooroo/roo-ui/commit/b922a8a))




<a name="0.7.0"></a>
# [0.7.0](https://github.com/hooroo/roo-ui/compare/v0.6.4...v0.7.0) (2018-05-11)


### Features

* add new icon set ([#14](https://github.com/hooroo/roo-ui/issues/14)) ([377c4c8](https://github.com/hooroo/roo-ui/commit/377c4c8))




<a name="0.5.4"></a>
## [0.5.4](https://github.com/hooroo/roo-ui/compare/v0.5.2...v0.5.4) (2018-05-08)


### Bug Fixes

* replace build script with prepublish ([90cafe3](https://github.com/hooroo/roo-ui/commit/90cafe3))




<a name="0.5.3"></a>
## [0.5.3](https://github.com/hooroo/roo-ui/compare/v0.5.2...v0.5.3) (2018-05-08)


### Bug Fixes

* replace build script with prepublish ([90cafe3](https://github.com/hooroo/roo-ui/commit/90cafe3))




<a name="0.5.0"></a>
# [0.5.0](https://github.com/hooroo/roo-ui/compare/v0.4.2...v0.5.0) (2018-05-07)


### Features

* add ui primitives ([#13](https://github.com/hooroo/roo-ui/issues/13)) ([5789b96](https://github.com/hooroo/roo-ui/commit/5789b96))




<a name="0.4.2"></a>
## [0.4.2](https://github.com/hooroo/roo-ui/compare/v0.4.0...v0.4.2) (2018-04-26)




**Note:** Version bump only for package @roo-ui/test-utils

<a name="0.4.1"></a>
## [0.4.1](https://github.com/hooroo/roo-ui/compare/v0.4.0...v0.4.1) (2018-04-26)




**Note:** Version bump only for package @roo-ui/test-utils

<a name="0.4.0"></a>
# [0.4.0](https://github.com/hooroo/roo-ui/compare/v0.3.3...v0.4.0) (2018-04-26)


### Features

* add theming ([bee896a](https://github.com/hooroo/roo-ui/commit/bee896a))
