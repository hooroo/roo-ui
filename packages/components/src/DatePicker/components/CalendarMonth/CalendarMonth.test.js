import React from 'react';
import { qantas as theme } from '@roo-ui/themes';
import { shallowWithTheme } from '@roo-ui/test-utils';
import { addDays } from 'date-fns';

import CalendarMonth from '.';

describe('<CalendarMonth />', () => {
  let wrapper;

  const startDate = new Date(2018, 7, 1, 10, 33, 30, 0);

  const props = {
    monthName: 'Jul',
    month: 7,
    year: 2018,
    stacked: true,
    weekdayNames: ['S', 'M', 'T', 'W', 'T', 'F', 'S'],
    weeks: [
      [
        { date: startDate },
        { date: addDays(startDate, 1) },
        { date: addDays(startDate, 2) },
        { date: addDays(startDate, 3) },
        { date: addDays(startDate, 4) },
        { date: addDays(startDate, 5) },
        { date: addDays(startDate, 7) },
      ],
      [
        { date: addDays(startDate, 7) },
        { date: addDays(startDate, 8) },
        { date: addDays(startDate, 9) },
        { date: addDays(startDate, 10) },
        { date: addDays(startDate, 11) },
        { date: addDays(startDate, 12) },
        { date: addDays(startDate, 14) },
      ]],
    getDateProps: jest.fn,
    isInRange: jest.fn,
    onMouseEnterOfDay: jest.fn,
  };

  beforeEach(() => {
    wrapper = shallowWithTheme(<CalendarMonth {...props}>Days</CalendarMonth>, theme);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('<CalendarWeekdays />', () => {
    it('renders CalendarWeekdays', () => {
      expect(wrapper.find('CalendarWeekdays').prop('month')).toEqual(props.month);
      expect(wrapper.find('CalendarWeekdays').prop('year')).toEqual(props.year);
      expect(wrapper.find('CalendarWeekdays').prop('weekdayNames')).toEqual(props.weekdayNames);
    });
  });

  describe('<CalendarDays />', () => {
    it('passes the correct props', () => {
      expect(wrapper.find('CalendarDays').prop('month')).toEqual(props.month);
      expect(wrapper.find('CalendarDays').prop('year')).toEqual(props.year);
      expect(wrapper.find('CalendarDays').prop('weeks')).toEqual(props.weeks);
      expect(wrapper.find('CalendarDays').prop('isInRange')).toEqual(props.isInRange);
      expect(wrapper.find('CalendarDays').prop('onMouseEnterOfDay')).toEqual(props.onMouseEnterOfDay);
    });
  });
});
