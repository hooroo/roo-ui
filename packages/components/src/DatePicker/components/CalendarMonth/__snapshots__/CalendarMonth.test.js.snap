// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalendarMonth /> renders correctly 1`] = `
<Box
  blacklist={
    Array [
      "m",
      "mt",
      "mr",
      "mb",
      "ml",
      "mx",
      "my",
      "p",
      "pt",
      "pr",
      "pb",
      "pl",
      "px",
      "py",
      "width",
      "w",
      "minWidth",
      "maxWidth",
      "height",
      "minHeight",
      "maxHeight",
      "color",
      "bg",
      "textAlign",
      "align",
      "boxShadow",
      "borderRadius",
      "border",
      "borderTop",
      "borderRight",
      "borderBottom",
      "borderLeft",
      "borderColor",
      "flexBasis",
      "flex",
      "position",
      "zIndex",
      "monthsToDisplay",
      "stacked",
    ]
  }
  monthsToDisplay={1}
  stacked={true}
>
  <Text
    blacklist={
      Array [
        "textStyle",
        "color",
        "bg",
        "fontSize",
        "f",
        "fontWeight",
        "letterSpacing",
        "lineHeight",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "textAlign",
        "align",
        "textDecoration",
        "hidden",
      ]
    }
    hidden={false}
    textStyle="caps"
  >
    Jul
     
    2018
  </Text>
  <CalendarWeekdays
    month={7}
    weekdayNames={
      Array [
        "S",
        "M",
        "T",
        "W",
        "T",
        "F",
        "S",
      ]
    }
    year={2018}
  />
  <CalendarDays
    disabledDates={Array []}
    getDateProps={[Function]}
    interactiveDisabledDates={false}
    isInRange={[Function]}
    month={7}
    onMouseEnterOfDay={[Function]}
    weeks={
      Array [
        Array [
          Object {
            "date": 2018-08-01T00:33:30.000Z,
          },
          Object {
            "date": 2018-08-02T00:33:30.000Z,
          },
          Object {
            "date": 2018-08-03T00:33:30.000Z,
          },
          Object {
            "date": 2018-08-04T00:33:30.000Z,
          },
          Object {
            "date": 2018-08-05T00:33:30.000Z,
          },
          Object {
            "date": 2018-08-06T00:33:30.000Z,
          },
          Object {
            "date": 2018-08-08T00:33:30.000Z,
          },
        ],
        Array [
          Object {
            "date": 2018-08-08T00:33:30.000Z,
          },
          Object {
            "date": 2018-08-09T00:33:30.000Z,
          },
          Object {
            "date": 2018-08-10T00:33:30.000Z,
          },
          Object {
            "date": 2018-08-11T00:33:30.000Z,
          },
          Object {
            "date": 2018-08-12T00:33:30.000Z,
          },
          Object {
            "date": 2018-08-13T00:33:30.000Z,
          },
          Object {
            "date": 2018-08-15T00:33:30.000Z,
          },
        ],
      ]
    }
    year={2018}
  />
</Box>
`;
